### PR TITLE
#2920 HttpContentDecompressor Compression level becomes more lenient fashion

### LINKED ISSUE #2920: Vertx-web-client  setTryUseCompression error:decompression failure
hello,
I use webclient request "http://api.bilibili.com/x/v1/dm/list.so?oid=87113082"

code :
```
val userAgent = "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36"
    val webClientOptions = webClientOptionsOf(userAgent=userAgent,followRedirects = true,tryUseCompression = true)
    val weblient = WebClient.create(vertx,webClientOptions)
val response = webClient
        .get(bilibiliApiUrl, "/x/v1/dm/list.so")
        .putHeader("Accept-Encoding","gzip, deflate")
        .addQueryParam("oid", cid.toString())
        .addQueryParam("page", i.toString())
        .sendAwait()
```
But there was an error：
```

io.netty.handler.codec.compression.DecompressionException: decompression failure
        at io.netty.handler.codec.compression.JdkZlibDecoder.decode(JdkZlibDecoder.java:230)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.embedded.EmbeddedChannel.writeInbound(EmbeddedChannel.java:325)
        at io.netty.handler.codec.http.HttpContentDecoder.decode(HttpContentDecoder.java:234)
        at io.netty.handler.codec.http.HttpContentDecoder.decodeContent(HttpContentDecoder.java:155)
        at io.netty.handler.codec.http.HttpContentDecoder.decode(HttpContentDecoder.java:147)
        at io.netty.handler.codec.http.HttpContentDecoder.decode(HttpContentDecoder.java:46)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
        at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.util.zip.DataFormatException: incorrect header check
        at java.base/java.util.zip.Inflater.inflateBytesBytes(Native Method)
        at java.base/java.util.zip.Inflater.inflate(Inflater.java:378)
        at io.netty.handler.codec.compression.JdkZlibDecoder.decode(JdkZlibDecoder.java:187)
        ... 38 more

```

### PR DESCRIPTION
**Pull request by shenzhou-6**
Fixes #2920

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ private void applyHttp1xConnectionOptions(ChannelPipeline pipeline) {
       false,
       options.getDecoderInitialBufferSize()));
     if (options.isTryUseCompression()) {
-      pipeline.addLast("inflater", new HttpContentDecompressor(true));
+      pipeline.addLast("inflater", new HttpContentDecompressor());
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java]: can we use `false` instead to be explicit and not rely on the default constructor ?

**Comment by vietj** [comment]
this also happens in `HttpServerImpl` so can this PR handle and test both cases to solve all problems at once ?

File: src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java (hunk 1 above)
Comment (shenzhou-6) [reply src/main/java/io/vertx/core/http/impl/HttpChannelConnector.java]: I think you are right,This place should not depend on default。

**Comment by shenzhou-6** [comment]
> this also happens in `HttpServerImpl` so can this PR handle and test both cases to solve all problems at once ?

I ignored HttpServerImpl
I will submit about httpServerImpl later.

**Comment by shenzhou-6** [comment]
I ignored the problem with HttpServerImpl, so I wrote some unnecessary code in the test case.
In fact, when the compression of HttpServerImpl is also loose. There will be no such problems.

So in the test case, as long as the server and client open a relatively loose compression method at the same time
To compare values

**Comment by vietj** [comment]
can you also add a test for server decompression behavior ?

**Pull request merged**